src/plugins/vctlGit.nim

```diff
@@ -191,17 +204,126 @@ proc findGitDir(fullpath: string): string =
 # Using this in the GitRepo plugin too.
 type
   RepoInfo = ref object
-    vcsDir:   string
-    origin:   string
-    branch:   string
-    commitId: string
+    vcsDir:     string
+    origin:     string
+    branch:     string
+    commitId:   string
+    author:     string
+    authorDate: string
+    committer:  string
+    commitDate: string
 
   GitInfo = ref object of RootRef
     branchName: Option[string]
     commitId:   Option[string]
     origin:     Option[string]
     vcsDirs:    OrderedTable[string, RepoInfo]
 
+proc getUint32BE(data: string, whence: SomeInteger=0): uint32 =
+  result = cast[ref [uint32]](addr data[whence])[]
```

Comment:
The [] around uint32 isn't needed. I also tend to use `ptr` here instead of `ref` though they both work fine.

Reply:
Also, the following works:
```
import nativesockets
var be = htonl(x)
```

Reply:
for `ref` vs `ptr` I was just using code I already had handy since nim is still a bit foreign for me, i'll change to `ptr`. I figured there was a `htonl` somewhere but digging felt like more work than shifting bits :). I'll change that too